### ISSUE TITLE
OAuth / Graph API

### ISSUE THREAD

**Issue by SafariKC** [martijndierckx/homebridge-hydrawise#25]
Looks like they kicked their Home Automation possibilities up a notch with a new Graph based API that uses OAuth for access.  I don't see any public docs but might be interesting to ask them about this option?

**Comment by SafariKC** [comment]
The Graph QL & oAuth 2.0 API is ideal for commercial applications, home automation, and government agencies.
It is rate limited and provides a secure GDPR & CCPA compliant API. It provides the following information:
Controller names
Zone number and name
Time until next run
Currently running
Length of run time
Sensor status
Manual start and stop
Manual run all stations Suspend a zone or all zones Unsuspend a zone or all zones
If you have additional questions regarding the API please email anthony.long@hunterindustries.com.

**Comment by martijndierckx** [comment]
As far as I can read from their very limited documentation, the only extra the GraphQL API supports are getting sensor values. Do you see any others?

If there are enough votes on this topic, I will invest some time in switching API types.

**Comment by SafariKC** [comment]
Yes, I agree the documentation is limited.  But the way I read this is that the GraphQL would allow you to query the graph which wouldn't have the limits that the direct API via REST would have.   https://www.apollographql.com/blog/graphql-vs-rest-5d425123e34b/

**Comment by martijndierckx** [comment]
They're still mentioning a rate limit on the limited documentation page. But they do not mention what the rate is anymore...

**Comment by SafariKC** [comment]
Yeah... there's an email address there.. so I may ping them to find out what the deal is.  But this must be what they are replacing Local with.

**Comment by clashcaddie** [comment]
Ive been looking into the GraphQL today and it doesn't seem to be rate limited.

I will work on adding support for this over the coming weeks.

**Comment by anthonyvallone** [comment]
Ive been looking into the GraphQL today and it doesn't seem to be rate limited.

I will work on adding support for this over the coming weeks.

**Comment by martijndierckx** [comment]
Maybe contact their support to check, before investing to much time. But you’re very welcome to contribute!
________________________________
Van: Anthony ***@***.***>
Verzonden: Sunday, February 12, 2023 4:20:52 AM
Aan: martijndierckx/homebridge-hydrawise ***@***.***>
CC: Martijn Dierckx ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [martijndierckx/homebridge-hydrawise] OAuth / Graph API (#25)


Ive been looking into the GraphQL today and it doesn't seem to be rate limited.

I will work on adding support for this over the coming weeks.
